src/test/test-Stream.ts

```diff
@@ -2084,4 +2085,312 @@ describe("Stream", () => {
 			}, 10);
 		});
 	});
+
+	describe("batch()", () => {
+		const batchResults: number[][] = [];
+		beforeEach(() => batchResults.splice(0));
+
+		function pipeWithDelay(
```

Comment:
Ironically given our last conversation about `abort()`, I'm now finding it difficult to write sources that *don't* reject on abort. The awkward code here was an attempt to avoid the default behaviour of `map()` which was causing the stream result to reject.

I'll just rewrite the abort test to represent a more typical abort-as-failure-state scenario, I think.